src/analysis/normed_space/star/continuous_functional_calculus.lean

```diff
@@ -0,0 +1,275 @@
+  exact ennreal.coe_lt_coe.1
+  (calc (‖star a' * a' - algebra_map ℂ _ (‖star a * a‖)‖₊ : ℝ≥0∞)
+      = ‖algebra_map ℂ A (‖star a * a‖) - star a * a‖₊ : by { rw [←nnnorm_neg, neg_sub], refl }
+  ... = spectral_radius ℂ (algebra_map ℂ A (‖star a * a‖) - star a * a)
+      : begin
+          refine (is_self_adjoint.spectral_radius_eq_nnnorm _).symm,
+          rw [is_self_adjoint, star_sub, star_mul, star_star, ←algebra_map_star_comm,
+            is_R_or_C.star_def, is_R_or_C.conj_of_real],
+        end
+  ... < ‖star a * a‖₊ : spectrum.spectral_radius_lt_of_forall_lt _ h₂ ),
+end
+
+/-- For `a : A` which is invertible in `A`, the inverse lies in any unital C⋆-subalgebra `S`
+containing `a`. -/
+lemma star_subalgebra.is_unit_coe_inv_mem {S : star_subalgebra ℂ A} (hS : is_closed (S : set A))
+  {x : A} (h : is_unit x) (hxS : x ∈ S) : ↑h.unit⁻¹ ∈ S :=
+begin
+  have hx := h.star.mul h,
+  suffices this : (↑hx.unit⁻¹ : A) ∈ S,
+  { rw [←one_mul (↑h.unit⁻¹ : A), ←hx.unit.inv_mul, mul_assoc, is_unit.unit_spec, mul_assoc,
+      h.mul_coe_inv, mul_one],
+    exact mul_mem this (star_mem hxS) },
+  refine le_of_is_closed_of_mem ℂ hS (mul_mem (star_mem hxS) hxS) _,
+  haveI := (is_self_adjoint.star_mul_self x).is_star_normal,
+  have hx' := elemental_star_algebra.is_unit_of_is_unit_of_is_star_normal hx,
+  convert (↑hx'.unit⁻¹ : elemental_star_algebra ℂ (star x * x)).prop using 1,
+  exact left_inv_eq_right_inv hx.unit.inv_mul (congr_arg coe hx'.unit.mul_inv),
+end
+
+/-- For a unital C⋆-subalgebra `S` of `A` and `x : S`, if `↑x : A` is invertible in `A`, then
+`x` is invertible in `S`. -/
+lemma star_subalgebra.coe_is_unit {S : star_subalgebra ℂ A} (hS : is_closed (S : set A)) {x : S} :
+  is_unit (x : A) ↔ is_unit x :=
+begin
+  refine ⟨λ hx, ⟨⟨x, ⟨(↑hx.unit⁻¹ : A), star_subalgebra.is_unit_coe_inv_mem hS hx x.prop⟩, _, _⟩,
+    rfl⟩, λ hx, hx.map S.subtype⟩,
+  exacts [subtype.coe_injective hx.mul_coe_inv, subtype.coe_injective hx.coe_inv_mul],
+end
+
+lemma star_subalgebra.mem_spectrum_iff {S : star_subalgebra ℂ A} (hS : is_closed (S : set A))
+  {x : S} {z : ℂ} : z ∈ spectrum ℂ x ↔ z ∈ spectrum ℂ (x : A) :=
+not_iff_not.2 (star_subalgebra.coe_is_unit hS).symm
+
+/-- **Spectral permanence.** The spectrum of an element is invariant of the (closed)
+`star_subalgebra` in which it is contained. -/
+lemma star_subalgebra.spectrum_eq {S : star_subalgebra ℂ A} (hS : is_closed (S : set A)) (x : S) :
+  spectrum ℂ x = spectrum ℂ (x : A) :=
+set.ext $ λ z, star_subalgebra.mem_spectrum_iff hS
+
+variables (a)
+
+/-- The natural map from `character_space ℂ (elemental_star_algebra ℂ a)` to `spectrum ℂ a` given
+by evaluating `φ` at `a`. This is essentially just evaluation of the `gelfand_transform` of `a`,
+but because we want something in `spectrum ℂ a`, as opposed to
+`spectrum ℂ ⟨a, elemental_star_algebra.self_mem ℂ a⟩` there is slightly more work to do. -/
+@[simps]
+noncomputable def elemental_star_algebra.character_space_to_spectrum (x : A) :
```

Comment:
The continuity of this map will be superseded by the fact that it's a homeo, proved below. So I dont think the bundled continuous map is useful. I'd rather have the unbundled version first, and then a lemma saying that it is continuous (or put the proof of continuity inside the proof of `elemental_star_algebra.character_space_homeo`)